Musl support

Hello! Today I encountered scx compilation and received an error

```rust
error[E0063]: missing fields `sched_ss_init_budget`, `sched_ss_low_priority`, `sched_ss_max_repl` and 1 other field in initializer of `sched_param`
    --> src/bpf.rs:377:34
     |
377 |          let param: sched_param = sched_param { sched_priority: 0 };
     |                                   ^^^^^^^^^^^ missing `sched_ss_init_budget`, `sched_ss_low_priority`, `sched_ss_max_repl` and 1 other field
```

Understanding the problem, I discovered that the libc library, in the presence of musl, [has additional fields in sched_param](https://github.com/rust-lang/libc/blob/acc7bb1a38e2b9d6a18f28279d80d464c4c25089/src/unix/linux_like/mod.rs#L105-L115)

--- a/rust/scx_rustland_core/assets/bpf.rs
+++ b/rust/scx_rustland_core/assets/bpf.rs
@@ -355,7 +355,22 @@ impl<'cb> BpfScheduler<'cb> {
     // Set scheduling class for the scheduler itself to SCHED_EXT
     fn use_sched_ext() -> i32 {
         let pid = std::process::id();
+        #[cfg(target_env = "gnu")]
         let param: sched_param = sched_param { sched_priority: 0 };
+        #[cfg(target_env = "musl")]
+        let param: sched_param = sched_param {
+            sched_priority: 0,
+            sched_ss_low_priority: 0,
+            sched_ss_repl_period: timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            },
+            sched_ss_init_budget: timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            },
+            sched_ss_max_repl: 0,
+        };
         let res =
             unsafe { sched_setscheduler(pid as i32, SCHED_EXT, &param as *const sched_param) };
         res

